scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala

```diff
@@ -1,206 +1,115 @@
-    val secretsManagerClient = mockSecretsManagerClient(Map("token" -> Nil))
+    val secret = Secret(Map("token" -> Nil))
 
-    val dependencies = Dependencies(secretId => IO.pure(userClient), secretId => secretsManagerClient)
+    val (_, _, res) = runLambda(rotationEvent, secret, Credentials("", ""))
 
-    val ex = new Lambda().handler(rotationEvent, config, dependencies).attempt.unsafeRunSync().left.value
-
-    ex.getMessage should equal("Secret version token not set as AWSPENDING for rotation of secret id.")
+    res.left.value.getMessage should equal("Secret version token not set as AWSPENDING for rotation of secret id.")
   }
 
-  "handler createSecret" should "return an error if the current secret doesn't exist" in {
+  "handler createSecret" should "return an error if there is an error getting the secret" in {
     val rotationEvent = RotationEvent(CreateSecret, "id", "token")
-    val config = Config("http://localhost")
-    val userClient = mock[UserClient[IO]]
-    val secretsManagerClient = mockSecretsManagerClient()
-
-    when(secretsManagerClient.getSecretValue[String](any[Stage])(using any[Decoder[String]]))
-      .thenReturn(IO.raiseError(new Exception("Cannot find secret id")))
-    val dependencies = Dependencies(secretId => IO.pure(userClient), secretId => secretsManagerClient)
+    val secret = Secret(Map("token" -> List(SecretStage(Map.empty, Pending))))
 
-    val ex = new Lambda().handler(rotationEvent, config, dependencies).attempt.unsafeRunSync().left.value
+    val (_, _, res) = runLambda(rotationEvent, secret, Credentials("", ""), Errors(getSecret = true).some)
 
-    ex.getMessage should equal("Cannot find secret id")
+    res.left.value.getMessage should equal("Error getting secret")
   }
 
   "handler createSecret" should "not call putSecretValue if a Pending value already exists" in {
     val rotationEvent = RotationEvent(CreateSecret, "id", "token")
-    val config = Config("http://localhost")
-    val userClient = mock[UserClient[IO]]
-    val secretsManagerClient = mockSecretsManagerClient()
+    val secret = Secret(Map("token" -> List(SecretStage(Map.empty, Pending))))
 
-    when(secretsManagerClient.getSecretValue[Map[String, String]](any[Stage])(using any[Decoder[Map[String, String]]]))
-      .thenReturn(IO(Map("user" -> "secret")))
-    when(secretsManagerClient.getSecretValue[Map[String, String]](any[String], any[Stage])(using any[Decoder[Map[String, String]]]))
-      .thenReturn(IO(Map("user" -> "pendingSecret")))
-    val dependencies = Dependencies(secretId => IO.pure(userClient), secretId => secretsManagerClient)
+    val (secretResult, _, _) = runLambda(rotationEvent, secret, Credentials("", ""))
 
-    new Lambda().handler(rotationEvent, config, dependencies).unsafeRunSync()
-
-    verify(secretsManagerClient, times(0)).putSecretValue(any[String], any[Stage], any[Option[String]])(using any[Encoder[String]])
+    secretResult.versionToStage.size should equal(1)
```

Comment:
Is it worth checking the password is still the same/the expected one?

Reply:
It's slightly superfluous but one extra line doesn't hurt. I've added it in.